backend/src/routers/user_auth/register.py

```diff
@@ -0,0 +1,57 @@
+from fastapi import HTTPException, APIRouter
+from starlette.status import HTTP_417_EXPECTATION_FAILED, HTTP_409_CONFLICT
+from starlette.responses import Response
+from pydantic import BaseModel
+from .utils import MOCK_DB, get_password_hash, User
+
+
+router = APIRouter()
+
+
+class Register_form(BaseModel):
+    username: str
+    email: str
+    password: str
+    rePassword: str
+
+
+def check_user_uniqueness(db, user: User):
+    # Needs refractoring for less stupid solution.
+    for users in db:
+        if (
+            db[users]["email"] == user["email"]  # pylint: disable=C0330
+            or db[users]["username"] == user["username"]  # pylint: disable=C0330
+        ):
+            return False
+    return True
+
+
+def add_user_to_db(user: User):
+    # There will be adding to database, but db is required.
+    print("User: ", user["username"], "added.")
+
+
+@router.post("/register")
+async def register(form_data: Register_form):
+    if form_data.password != form_data.rePassword:
+        print("Passwords were not the same.")
+        raise HTTPException(
+            status_code=HTTP_417_EXPECTATION_FAILED,
+            detail="Repeat password correctly.",
+        )
+
+    hashed_password = get_password_hash(form_data.password)
+    user = {
+        "username": form_data.username,
+        "email": form_data.email,
+        "hashed_password": hashed_password,
+        "disabled": False,
+    }
+    if not check_user_uniqueness(MOCK_DB, user):
+        print("User already taken.")
+        raise HTTPException(
+            status_code=HTTP_409_CONFLICT, detail="Username or email already taken.",
```

Comment:
This status code is also not an ideal use case. A simple bad request works better here too. You would want to use `409` in case of a conflict that you'd need to resolve in some other way and that would allow you to send the same request and make it pass with 200. This is not the case here, once a username is taken the user has no means of conflict resolution, he will have to change the username and retry the request with different parameters.

Reply:
Changed it, thanks for use case example, I'll have to do some more digging on http protocols.

Reply:
Most of the status codes are used in very specific use cases. Most of the time 400 to 404 codes are the best to use, but can't go wrong with any of them as long as the error is clear and descriptive.